.github/workflows/workflow.yml

```diff
@@ -0,0 +1,32 @@
+name: Continous integration tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```

Comment:
Can you give me a use case? I don't quite understand how this would be used 😄

Reply:
Sure, here's an example from my previous $job, but you can decide whether you think it'd be useful or not (I suspect you'd be the main user of this for now).

I used this in Cylc UI, a Vue.js app, in the `CI` GitHub action <https://github.com/cylc/cylc-ui/blob/3bc7ea7823389b712af4e59364cf3d03156edb26/.github/workflows/main.yml#L7>. With that when I click on the tab for the GitHub Actions in the repository, there's a button that allows me to "Run the workflow", without having to commit anything. In the screenshot below, I clicked on the Run workflow button to show that I get a dialogue to choose the branch too (if you have that in your fork, you can trigger GH actions for your fork/branch without pushing commits)

![image](https://user-images.githubusercontent.com/304786/144814461-380c6831-0ef1-44fb-a5e6-744f18cae4b4.png)

That's useful when you have a GH Action that failed because of some service that was offline, or if you'd like to re-trigger to confirm if it's not some brittle test, or in case you haven't created a PR and maybe the current action settings restrict it from running branches in your fork.

Reply:
Thanks for explaining, this is great! I will keep this in mind!